internal/storage/fake/bucket.go

```diff
@@ -442,10 +442,7 @@ func (b *bucket) newReaderLocked(
 	// Find the object with the requested name.
 	index = b.objects.find(req.Name)
 	if index == len(b.objects) {
-		err = &gcs.NotFoundError{
-			Err: fmt.Errorf("object %s not found", req.Name),
-		}
-
+		err = storage.ErrObjectNotExist
```

Comment:
This loses information as this error can no longer convey the name of the missing object.

Reply:
Why do we need this change and isn't there any alternative ?

Reply:
Updated the error to wrap the storage.ErrObjectNotExist with additional context.

Reply:
We need to change this because we recieve storage.ErrObjectNotExist while creating a new reader for real bucket and fake bucket should also return same error.

Reply:
Closing this PR as it is better to map storage.ErrObjectNotExist to gcs.NotFoundError in bucket_handle.go